src/xmpp/xmpp-im/xmpp_mamtask.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * xmpp_mam.cpp - XEP-0313 Message Archive Management
+ * Copyright (C) 2024 mcneb10
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef XMPP_MAM_TASK_H
+#define XMPP_MAM_TASK_H
+
+#include <QDateTime>
+#include <QList>
+#include <QSharedDataPointer>
+#include <QString>
+#include <QStringList>
+#include <qdom.h>
+
+#include "xmpp/jid/jid.h"
+#include "xmpp_task.h"
+
+#include "xmpp_xdata.h"
+#include "xmpp_xmlcommon.h"
+
+class QDomElement;
+class QString;
+
+namespace XMPP {
+class Jid;
+
+class MAMTask : public Task {
+    Q_OBJECT
+public:
+    MAMTask(Task *);
+    ~MAMTask();
+
+    const QList<QDomElement> &archive() const;
+
+    void get(const Jid &j, const QDateTime &from = QDateTime(), const QDateTime &to = QDateTime(),
```

Comment:
maybe you should review it. If you add more filters it will turn out into too many parameters. not that convenient.
In case it will make more sense to put common data sets in structures and likely use overloads and different task types for different MAM requests.

Reply:
i was thinking that if i put this behind a new `MAMManager` class i would keep this here but abstract it away from the developer. It's nice to have a way to initialize everything in `Private` in one go

Reply:
till the moment the parameters become mutually exclusive. something to think about later

Reply:
i made seperate get methods for time and ID filtering. The API is a little questionable, let me know if you want me to change it